api4/post.go

```diff
@@ -175,10 +175,15 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	skipFetchThreads := r.URL.Query().Get("skipFetchThreads") == "true"
 	collapsedThreads := r.URL.Query().Get("collapsedThreads") == "true"
 	collapsedThreadsExtended := r.URL.Query().Get("collapsedThreadsExtended") == "true"
+	includeDeleted := r.URL.Query().Get("includeDeleted") == "true"
 	channelId := c.Params.ChannelId
 	page := c.Params.Page
 	perPage := c.Params.PerPage
 
+	if !c.IsSystemAdmin() {
+		includeDeleted = false
+	}
+
```

Comment:
Should this return an unathorized response? 
Giving a result by changing the value might cause difficult to detect bugs in any tool that it is not aware of the behaviour.

Reply:
thank you for taking the time to review this PR.
I agree this should return an unauthorized or not allowed response. I have already updated this change.